src/app/src/main/java/com/example/myapplication/WelcomeActivity.java

```diff
@@ -28,20 +37,70 @@ protected void onCreate(Bundle savedInstanceState) {
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
             return insets;
         });
-        btn=findViewById(R.id.welcome_button);
-        btn.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
 
+        btn = findViewById(R.id.welcome_button);
+        btn.setVisibility(View.GONE);
 
-                Intent i = new Intent(WelcomeActivity.this, SignUpActivity.class);
-                startActivity(i);
-                finish();
+        dbManager = new DatabaseManager();
 
+        String deviceId = Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
 
-            }
+        dbManager.getUser(deviceId, this);
 
-        });
+        FirebaseFirestore db = FirebaseFirestore.getInstance();
+
+        db.collection("users").document(deviceId).get()
```

Comment:
this code is still here even though the DatabaseManager stuff has now been added?

Reply:
I'm just confused why there's both now?

Reply:
wait thats weird, i got rid of it
lemme try pushing again

Reply:
should be fixed now